### PR TITLE
Automated Test for Spanish Checkout, ASMFT-400

### PR DESCRIPTION
**Pull request by roppel5ls**
Adding an automated test for comprar.legalshield. Will need to maintain as development is not done and this is set up in the staging enviroment.

### CODE REVIEW

File: demo/SPANISH_regressionSuite/spanish_checkout.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  if (browser.desiredCapabilities.browserName = 'chrome'){
+    if (browser.desiredCapabilities.browserPlatform = 'Windows 10'){
+        click('button.ge-btn:nth-child(6)') //clicking submit
+        browser.pause(3000)
+        // waiting for confirmation page to load
+        browser.waitUntil(function (){
+          return browser.getUrl() === 'https://comprar.staging.legalshield.com/confirmation'},
+        20000)
+        browser.pause(5000)
+        assertText('body > div.ng-scope > section > div.confirmation__provider.ng-scope > div > div:nth-child(4) > div:nth-child(1) > div > div:nth-child(3) > p:nth-child(1)', spanishProvider)
```
Comment (TrentonWalkup) [inline demo/SPANISH_regressionSuite/spanish_checkout.js]: if you CAN, we might want a better selector here

File: demo/SPANISH_regressionSuite/spanish_checkout.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  enterText('//input[@name="bank_transit_number"]', '103101026')
+  enterText('//input[@name="bank_account_number"]', '1234567890123467')
+  browser.pause(5000)
+  assertValue('//input[@name="bank_name"]', bankName)
+  click('//div[7]/md-checkbox/div[1]')
+  scroll('//*[@id="main.paymentBDForm__next"]')
+  browser.pause(5000)
+  click('md-checkbox.ng-empty > div:nth-child(2)')
+  if (_.includes(notOffered,state) === true){
+    assertText('.sidebar__footer > footer:nth-child(2) > div:nth-child(1) > span:nth-child(2)', corpPrice)
```
Comment (TrentonWalkup) [inline demo/SPANISH_regressionSuite/spanish_checkout.js]: and here

File: demo/SPANISH_regressionSuite/spanish_checkout.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  } else {
+    assertText('.sidebar__footer > footer:nth-child(2) > div:nth-child(1) > span:nth-child(2)', comboBDTotal)
+  }
+  if (browser.desiredCapabilities.browserName = 'chrome'){
```
Comment (roppel5ls) [inline demo/SPANISH_regressionSuite/spanish_checkout.js]: I need to refactor from here down

Comment (roppel5ls) [reply demo/SPANISH_regressionSuite/spanish_checkout.js]: my if's at least

File: demo/SPANISH_regressionSuite/spanish_checkout.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  click('md-checkbox.ng-empty > div:nth-child(2)')
+  assertText('.sidebar__footer > footer:nth-child(2) > div:nth-child(1) > span:nth-child(2)', corpPrice)
+  if (browser.desiredCapabilities.browserName = 'chrome'){
+    if (browser.desiredCapabilities.browserPlatform = 'Windows 10'){
+        click('button.ge-btn:nth-child(6)') //clicking submit
+        browser.pause(3000)
+        // waiting for confirmation page to load
+        browser.waitUntil(function (){
+          return browser.getUrl() === 'https://comprar.staging.legalshield.com/confirmation'},
+        20000)
```
Comment (TrentonWalkup) [inline demo/SPANISH_regressionSuite/spanish_checkout.js]: 20 seconds seems long but i guess it is what it is

File: demo/SPANISH_regressionSuite/spanish_checkout.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
+var i = require('../tools.js')
```
Comment (TrentonWalkup) [inline demo/SPANISH_regressionSuite/spanish_checkout.js]: const please here and below

File: demo/SPANISH_regressionSuite/spanish_checkout.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+var _ = require('lodash')
+
+var corpPrice = '$24.95'
+var comboBDTotal = '$49.90'
+var bankName = 'VISION BANK NA'
+var spanishProvider = 'Spanish Legal Group, LLC'
+
```
Comment (TrentonWalkup) [inline demo/SPANISH_regressionSuite/spanish_checkout.js]: spacing?

File: demo/SPANISH_regressionSuite/spanish_checkout.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    browser.pause(3000)
+    click('.identity__header')
+    browser.pause(3000)
+    click('md-radio-button[aria-label="IDShield Family  ($24.95 por mes)"]')
+    click('.btn__addPlan')
+  }
+  browser.pause(3000)
+  click('button.ge-btn:nth-child(3)')
+  browser.pause(10000)
+  enterText('//input[@name="username"]', i.user.username)
```
[the remark this answers is not in the document]
Comment (roppel5ls) [reply demo/SPANISH_regressionSuite/spanish_checkout.js]: If I can get a story to create this function I would be happy to do that, then refactor the code to use it!

File: demo/SPANISH_regressionSuite/spanish_checkout.js (hunk 4 above)
Comment (roppel5ls) [reply demo/SPANISH_regressionSuite/spanish_checkout.js]: In production we semi-commonly see some API calls take upwards of 30 seconds to complete, I figured 20 was a good middle ground between what performance-wise it should be and what we see in production. I can definitely lower this if needed.

File: demo/SPANISH_regressionSuite/spanish_checkout.js (hunk 5 above)
Comment (roppel5ls) [reply demo/SPANISH_regressionSuite/spanish_checkout.js]: fixing

File: demo/SPANISH_regressionSuite/spanish_checkout.js (hunk 6 above)
Comment (roppel5ls) [reply demo/SPANISH_regressionSuite/spanish_checkout.js]: fixing

File: demo/SPANISH_regressionSuite/spanish_checkout.js (hunk 1 above)
Comment (roppel5ls) [reply demo/SPANISH_regressionSuite/spanish_checkout.js]: Would you mind briefly working with me this afternoon to use the .element() function? I tried yesterday to no avail..

**Comment by Gregorma** [comment]
LGTM

**Comment by TrentonWalkup** [comment]
LGTM

**Comment by Gregorma** [comment]
L
G
T
M

**Pull request merged**